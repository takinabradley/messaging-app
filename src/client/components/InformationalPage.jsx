import { Link } from "react-router-dom"
export default function InformationalPage() {
  return (
    <div>
      <div>Main page!</div>

      <div>stuff about this app!</div>

      <div>
        Use these links to <Link to="/login">login</Link> or{" "}
        <Link to="/register">register</Link>!
      </div>
    </div>
  )
}
