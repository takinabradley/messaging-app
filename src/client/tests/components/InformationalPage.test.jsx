import { describe, expect, it, test } from "vitest";
import InformationalPage from "../../components/InformationalPage";
import { render, screen } from "@testing-library/react";
import { createMemoryRouter, Outlet, RouterProvider } from "react-router-dom";
import userEvent from "@testing-library/user-event";

describe("AccountPage", () => {
  test("renders", async () => {
    const router = createMemoryRouter([
        {
            path: "/",
            element: <InformationalPage />
        },
    ])
    render(<RouterProvider router={router}/>)

    expect(screen.getByTestId("InformationalPage")).toBeInTheDocument()    
    expect(screen.getByTestId("InformationalPage")).toMatchSnapshot()
  })

  test("Login link directs to /login route", async () => {
    const user = userEvent.setup()
    const router = createMemoryRouter([
        {
            path: "/",
            element: <InformationalPage />
        },
        {
            path: "/login",
            element: "This is the login page"
        },
    ])

    render(<RouterProvider router={router}/>)

    const loginLink = screen.getByRole("link", {name: "login"})
    await user.click(loginLink)

    expect(screen.getByText("This is the login page")).toBeInTheDocument()    
  })

  test("register link directs to /register route", async () => {
    const user = userEvent.setup()
    const router = createMemoryRouter([
        {
            path: "/",
            element: <InformationalPage />
        },
        {
            path: "/register",
            element: "This is the register page"
        }
    ])

    render(<RouterProvider router={router}/>)

    const registerLlink = screen.getByRole("link", {name: "register"})
    await user.click(registerLlink)

    expect(screen.getByText("This is the register page")).toBeInTheDocument()    
  })
})
