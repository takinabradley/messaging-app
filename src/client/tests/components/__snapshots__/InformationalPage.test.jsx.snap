// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountPage > renders 1`] = `
<div
  class="InformationalPage"
  data-testid="InformationalPage"
>
  <div>
    Main page!
  </div>
  <div>
    stuff about this app!
  </div>
  <div>
    Use these links to 
    <a
      href="/login"
    >
      login
    </a>
     or
    <a
      href="/register"
    >
      register
    </a>
    !
  </div>
</div>
`;
